Grey Tide event light behavior changes and code tidiness

Rhials opened the pull request:
## About The Pull Request

Makes a number of changes to the Grey Tide event, some of which are and some of which aren't player facing.

Player facing bits:

- The lights will once again turn off when the event ends and the doors begin opening. Originally, the lights would all shatter when the event hit, providing a cover of darkness for opportunists to sneak around under. Unfortunately this would prevent the AI from fixing the doors until the lights were replaced, leading to headaches and the removal of this functionality in #43159. Now, the lights are simply turned off at the APC instead of shattered.
- The lights now flicker at certain intervals as the event is running, rather than just right at the start.
- Announcement has been slightly modified to indicate that the event hits more than just airlocks.
- Event runs way faster. Doors open all at once, rather than one-at-a-time.


And now for the non player-facing bits:

- The filename is now grey_tide.dm instead of prison_break.dm. The event was effectively re-branded to the Grey Tide event six years ago, but the filename was unchaged. 
- potential_areas is now a part of setup instead of a var on the round_event.
- Cleans up instances of single-character varnames and stuff not in snake case.
- The event now checks if it has any valid areas (will it work or not) at the start, rather than at the end.
- The event now sends a global signal when run, rather than checking everything in the entire world. It should run faster now.
- Everything that can be affected by the grey tide event now stores its functionality on a signal handler, including the light flickers

## Why It's Good For The Game

Makes the grey tide event a bit more interesting and easier to take advantage of. Cleans up the backend code a bit.

## Changelog
:cl: Rhials
balance: The lights will now flicker more frequently before a Grey Tide event
balance: The lights will now be shut off via the APC when the Grey Tide event opens up a department.
code: Makes some miscellaneous code improvements to the Grey Tide event
code: Grey tide event runs much faster
code: Renames prison_break.dm to grey_tide.dm
/:cl:

Rhials committed:
service is now targetable, also kills single char vars

Rhials committed:
service is now targetable, also kills single char vars

Rhials committed:
makes the light flickering last a bit longer

I didnt even know this made the lights flicker, wow!

Rhials committed:
makes the light flickering last a bit longer

I didnt even know this made the lights flicker, wow!

Rhials committed:
it begins...

Rhials committed:
it begins...

Rhials committed:
later

Rhials committed:
later

Rhials committed:
modifies light flicker behavior

also changes areasToOpen to areas_to_open

Rhials committed:
modifies light flicker behavior

also changes areasToOpen to areas_to_open

Rhials committed:
light switching

light switches now toggle off when the event occurs.

Does it work well? not really. A lot of places are just straight-up missing lightswitches. I'll have to test this more to see if it works fine on other maps

Rhials committed:
light switching

light switches now toggle off when the event occurs.

Does it work well? not really. A lot of places are just straight-up missing lightswitches. I'll have to test this more to see if it works fine on other maps

Rhials committed:
adds admin feedback and a better way of killing the lights

Rhials committed:
adds admin feedback and a better way of killing the lights

Rhials committed:
flicker() now respects area power

Rhials committed:
flicker() now respects area power

Rhials committed:
potential_areas is no longer a round_event var

it's only called once so it doesn't need to be stored or anything

Rhials committed:
potential_areas is no longer a round_event var

it's only called once so it doesn't need to be stored or anything

Rhials committed:
undo

Rhials committed:
undo

Rhials committed:
fuck you im not going to be responsible for this

Rhials committed:
fuck you im not going to be responsible for this

Rhials committed:
I think my branch might be outdated

Rhials committed:
I think my branch might be outdated

Rhials committed:
Merge branch 'I-never-watched-the-prison-breaktv-show-lol' of https://github.com/Rhials/tgstation into I-never-watched-the-prison-breaktv-show-lol

Rhials committed:
rebase complete (somehow), testing shows that nothing broke

officially renames the event file since it's no longer referred to as "prison break" anywhere

Rhials committed:
not sure how this survived the rebase

Rhials committed:
fuck it this will be in the next pr

Ryusenshu commented:
I have never seen Grey Tide shattering Lights, only flickering

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,68 @@
+	severity = rand(1,3)
+
+	var/list/potential_areas = list(/area/station/command,
+								/area/station/engineering,
+								/area/station/medical,
+								/area/station/security,
+								/area/station/cargo,
+								/area/station/science)
+
```
Only indent once, and format the end like this:

```dm
    /area/station/science,
)
```

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,68 @@
+				areas_to_open += area_to_check
+
+/datum/round_event/grey_tide/announce(fake)
+	priority_announce("Gr3y.T1d3 virus detected in [station_name()] secure locking encryption subroutines. Severity level of [severity]. Recommend station AI involvement.", "Security Alert") //It affects more than just doors!
```
```suggestion
	priority_announce("Gr3y.T1d3 virus detected in [station_name()] secure locking encryption subroutines. Severity level of [severity]. Recommend station AI involvement.", "Security Alert")
```

Historical comments are not helpful to anyone

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,68 @@
+		kill()
+
+/datum/round_event/grey_tide/tick()
+	if(ISMULTIPLE(activeFor, 12))
```
Early-return

Mothblocks replied:
`if(!ISMULTIPLE(activeFor, 12)) return`

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,68 @@
+	if(ISMULTIPLE(activeFor, 12))
+		for(var/area/area_to_open in areas_to_open)
+			for(var/obj/machinery/light/chosen_light in area_to_open)
+				chosen_light.flicker(10)
```
Named parameter for flicker. Also is this meant to be seconds?

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,68 @@
+
+/datum/round_event/grey_tide/end()
+	for(var/area/area_to_open in areas_to_open)
+		for(var/obj/object_to_open in area_to_open)
```
This is *extremely* slow.

Every `for (x in area)` loop goes over **every single thing in the entire world** to check if the area matches.

Instead of this, register a signal on all these objects to tell them how to react.

Rhials committed:
first fixes

Rhials committed:
Merge remote-tracking branch 'upstream/master' into I-never-watched-the-prison-breaktv-show-lol

Rhials committed:
signals wow

Rhials committed:
and it all actually works wow

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,51 @@
+
+/datum/round_event/grey_tide/end()
+	for(var/area/area_to_open in areas_to_open)
+		for(var/obj/object_to_open in area_to_open)
```
You still CANNOT do it like this. Looping over area is slow. You maybe just want to loop over all machines in GLOB.machines and then send signals to those. You can also send a global signal and inside the handlers check if their area is valid there.

LemonInTheDark commented:
BTW rather then looping every object in the world, you can use the lights list on area to iterate all of the area's lights.
for everything else use signals off the area, update them in the on changed area thing that exists somewhere

Wallemations commented:
Do the lights just turn off or do they have a set time where they're locked off?

Rhials committed:
starts moving stuff to global signals

the event no longer picks all subtypes of a station area and adds it to a list for comparison. Now it just takes the general area type and performs an istype check on the  signal handler proc.

This currently DOES NOT WORK because for some reason ssdcs is getting passed as an arg to the signals instead of the area subtype list.

I'm going to try something really stupid with this next commit that might work and might be useful for something I want to implement after this PR is done

Rhials committed:
greytide areas are now a global list to be read from

now for the lighting

Rhials committed:
dear diary, today I learned about global signals

wow

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -87,13 +87,18 @@
-	prison_open()
+	SIGNAL_HANDLER
+
+	if(is_station_level(z))
```
Early-return

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -87,13 +87,18 @@
+	SIGNAL_HANDLER
+
+	if(is_station_level(z))
+		for(var/area_type in GLOB.grey_tide_areas)
```
Early-continue

Mothblocks replied:
Looks fine

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -87,13 +87,18 @@
+			if(istype(get_area(src), area_type))
+				if(critical_machine) //Skip doors in critical positions, such as the SM chamber.
+					return
+				INVOKE_ASYNC(src, PROC_REF(prison_open), 2) //Sleep gets called further down in open(), so we have to invoke async
```
Named parameter for 2. Because this is INVOKE_ASYNC, it isn't supported, but you can do `/* name = */ 2`

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -26,26 +27,29 @@
-		for(var/area/area_to_open in areas_to_open)
+		for(var/area/area_to_open in GLOB.grey_tide_areas) //Currently broken due to the changes to areas_to_open. Doesn't matter because this functionality was getting thrown out anyways
 			for(var/obj/machinery/light/chosen_light in area_to_open)
 				chosen_light.flicker(10)
```
This isn't resolved, 10 needs to be a named parameter

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -26,26 +27,29 @@
 /datum/round_event/grey_tide/tick()
 	if(ISMULTIPLE(activeFor, 12))
-		for(var/area/area_to_open in areas_to_open)
+		for(var/area/area_to_open in GLOB.grey_tide_areas) //Currently broken due to the changes to areas_to_open. Doesn't matter because this functionality was getting thrown out anyways
```
Can you describe this more in detail in the comment? What is the problem and when are we removing this? Or is this your TODO?

Mothblocks commented on code/modules/power/apc/apc_main.dm:
```diff
@@ -621,9 +621,14 @@
-	update()
+	SIGNAL_HANDLER
+
+	if(is_station_level(z))
```
Early-return

Rhials committed:
early returns

Rhials committed:
Merge remote-tracking branch 'upstream/master' into I-never-watched-the-prison-breaktv-show-lol

Rhials committed:
whoosh

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -31,6 +39,21 @@
+/obj/machinery/door/airlock/proc/grey_tide()
+	SIGNAL_HANDLER
+
+	if(!is_station_level(z) || critical_machine) return //Skip doors in critical positions, such as the SM chamber.
```
Put `return` on its own line

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -31,6 +39,21 @@
+/obj/machinery/door/airlock/Destroy()
+	if(frequency)
+		SSradio.remove_object(src,frequency)
+	UnregisterSignal(SSdcs, COMSIG_GLOB_GREY_TIDE)
```
Signals are automatically cleared, you don't need to do this

Mothblocks commented on code/game/machinery/airlock_control.dm:
```diff
@@ -31,6 +39,21 @@
 	return ..()
 
+/obj/machinery/door/airlock/Destroy()
+	if(frequency)
```
You accidentally added this back?

Mothblocks commented on code/game/machinery/doors/brigdoors.dm:
```diff
@@ -261,6 +267,15 @@
+/obj/machinery/status_display/door_timer/proc/grey_tide()
+	SIGNAL_HANDLER
+
+	if(!is_station_level(z)) return
```
return on its own line

Mothblocks commented on code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm:
```diff
@@ -7,3 +7,21 @@
+/obj/structure/closet/secure_closet/proc/grey_tide()
+	SIGNAL_HANDLER
+
+	if(!is_station_level(z)) return
```
return on its own line

Mothblocks commented on code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm:
```diff
@@ -7,3 +7,21 @@
+
+/obj/structure/closet/secure_closet/Destroy()
+	. = ..()
+	UnregisterSignal(SSdcs, COMSIG_GLOB_GREY_TIDE)
```
Not necessary

Rhials committed:
sproink

Rhials committed:
you too

Mothblocks commented on code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm:
```diff
@@ -7,3 +7,18 @@
+	for(var/area_type in GLOB.grey_tide_areas)
+		if(istype(get_area(src), area_type))
+			locked = FALSE
+			update_appearance()
```
```suggestion
			update_appearance(UPDATE_ICON)
			return
```

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,61 @@
+	)
+
+	for(var/i in 1 to severity)
+		GLOB.grey_tide_areas += pick_n_take(potential_areas)
```
I don't think we should be putting this in a glob, we theoretically want to be able to support multiple at once.

Instead, create the list, and send that list through the signal.

Mothblocks replied:
Need to do this still

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,61 @@
+		kill()
+
+/datum/round_event/grey_tide/tick()
+	if(!ISMULTIPLE(activeFor, 12)) return
```
Two lines

Mothblocks commented on code/modules/events/grey_tide.dm:
```diff
@@ -0,0 +1,61 @@
+
+/datum/round_event/grey_tide/start()
+	if(!length(GLOB.grey_tide_areas))
+		log_world("ERROR: Could not initiate grey-tide. No areas in the list!")
```
```suggestion
		stack_trace("Could not initiate grey-tide. No areas in the list!")
```

Mothblocks commented on code/modules/power/apc/apc_main.dm:
```diff
@@ -615,6 +618,18 @@
+		return
+
+	for(var/area_type in GLOB.grey_tide_areas)
+		if(istype(get_area(src), area_type))
```
Do the same thing here, only UPDATE_ICON and `return` when you succeed.

Also all of these should probably be early-continued.

Mothblocks replied:
Just makes it look nicer, it's nice to enforce your invariants early

Rhials committed:
early returns, continues, changes

Rhials committed:
you too

Mothblocks reviewed (changes requested):
Do the list thing and this is ready

Rhials committed:
mastery

Mothblocks commented on code/__DEFINES/dcs/signals/signals_event.dm:
```diff
@@ -0,0 +1,6 @@
+// Grey Tide event signals
+/// Sent when the Grey Tide event begins affecting the station
```
```suggestion
/// Sent when the Grey Tide event begins affecting the station.
/// (list/area/greytide_areas)
```

Mothblocks committed:
Include arguments in comment
